tests/foreman/destructive/test_installer.py

```diff
@@ -147,3 +147,40 @@ def test_positive_installer_certs_regenerate(target_sat):
     )
     assert result.status == 0
     assert 'FAIL' not in target_sat.cli.Base.ping()
+
+
+def test_positive_installer_puma_worker_count(target_sat):
+    """Installer should set the puma worker count and thread max without having to manually
+    restart the foreman service.
+
+    :id: d0e7d958-dd3e-4962-bf5a-8d7ec36f3485
+
+    :steps:
+        1. Check how many puma workers there are by default (9)
+        2. Change answer's file to have 5 puma workers
+        3. Run satellite-installer --foreman-foreman-service-puma-workers 5 --foreman-foreman-service-puma-threads-max 5
+
+    :expectedresults: aux should show there are only 5 puma workers after installer runs
+
+    :BZ: 2025760
+
+    :customerscenario: true
+    """
+    result = target_sat.execute('ps aux | grep -v grep | grep -e USER -e puma')
+    for i in range(9):
```

Comment:
this feels a bit fragile towards future default value changes, would it make sense to first source the default setting from the answer file?

Reply:
The answer file is blank which is what makes this create 9 puma workers by default. We could instead count the amount of puma workers and then range off of the len?

Reply:
Ok so I learned this piece of code determines worker count https://github.com/theforeman/puppet-foreman/blob/908ed638790f3cfc6e2f5f62e1b023c907a9913c/manifests/config.pp#L83C1-L92. Therefore, I changed the test to record what the current worker count is, run the installer with worker count set to a number less than that count, and then assert it worked.